### PR TITLE
Opt-in enable_talkback

### PR DESCRIPTION
**Pull request by ktomala**
Allows enable_talkback in configuration, via commandline and RPC

### CODE REVIEW

File: golem/client.py [hunk 1]
```diff
@@ -1152,6 +1153,18 @@ def _unlock_datadir(self):
...
+    def enable_talkback(self, value):
+        talkback_value = bool(value)
+        logger_root = logging.getLogger()
+        try:
+            sentry_handler = [
+                h for h in logger_root.handlers if h.name == 'sentry'][0]
+            msg_part = 'Enabling' if talkback_value else 'Disabling'
+            log.info('{0} talkback service'.format(msg_part))
+            sentry_handler.set_enabled(talkback_value)
+        except Exception as e:
```
Comment (mfranciszkiewicz) [inline golem/client.py]: `# pylint: disable=broad-exception`

File: golem/rpc/mapping/rpcmethodnames.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
     get_setting=            'env.opt',
     update_setting=         'env.opt.update',
     get_datadir=            'env.datadir',
+    enable_talkback=        'env.enable_talkback',
```
Comment (mfranciszkiewicz) [inline golem/rpc/mapping/rpcmethodnames.py]: Since the config file isn't updated, please remove this mapping.

File: golem/client.py [hunk 3]
```diff
@@ -1152,6 +1153,19 @@ def _unlock_datadir(self):
             pass
         self.__datadir_lock.close()
 
+    def enable_talkback(self, value):   # pylint: disable=no-self-use
```
Comment (maaktweluit) [inline golem/client.py]: i think `@staticmethod` is better then disable pylint

File: golem/client.py [hunk 4]
```diff
@@ -1152,6 +1153,19 @@ def _unlock_datadir(self):
...
         self.__datadir_lock.close()
 
+    def enable_talkback(self, value):   # pylint: disable=no-self-use
+        talkback_value = bool(value)
+        logger_root = logging.getLogger()
+        try:
+            sentry_handler = [
+                h for h in logger_root.handlers if h.name == 'sentry'][0]
+            msg_part = 'Enabling' if talkback_value else 'Disabling'
+            logger.info('%s talkback service' % msg_part)
```
Comment (maaktweluit) [inline golem/client.py]: please use logger arguments over formatting wih `%`

File: golem/client.py [hunk 5]
```diff
@@ -1152,6 +1153,19 @@ def _unlock_datadir(self):
             pass
         self.__datadir_lock.close()
 
+    def enable_talkback(self, value):   # pylint: disable=no-self-use
+        talkback_value = bool(value)
+        logger_root = logging.getLogger()
+        try:
+            sentry_handler = [
+                h for h in logger_root.handlers if h.name == 'sentry'][0]
+            msg_part = 'Enabling' if talkback_value else 'Disabling'
```
Comment (maaktweluit) [inline golem/client.py:1169]: msg_part can be used in the exception log as well

**Review by maaktweluit — CHANGES_REQUESTED** [review]
LGTM!

Some small comments, will approve on reply

As general comment i do not see the value in `bool()` everywhere, if we make the default value False there is no other option. Saves a lot of type-casts and can be typed at the argument level

File: golem/client.py (hunk 3 above)
Comment (ktomala) [reply golem/client.py]: good point, will correct it

File: golem/client.py (hunk 4 above)
Comment (ktomala) [reply golem/client.py]: good find, will do

File: golem/client.py (hunk 5 above)
Comment (ktomala) [reply golem/client.py:1169]: also good point

**Review by maaktweluit — APPROVED** [review]

**Pull request merged**